openmc/mesh.py

```diff
@@ -1775,15 +1784,15 @@ def filename(self):
 
     @filename.setter
     def filename(self, filename):
-        cv.check_type('Unstructured Mesh filename', filename, (str, Path))
+        cv.check_type('Unstructured Mesh filename', filename, PathLike)
```

Comment:
```suggestion
        cv.check_type('Unstructured Mesh filename', filename, (str, Path))
```
Putting this back to the original pre PR version to avoid this error in CI
```TypeError: Subscripted generics cannot be used with class and instance checks```